driver-sync/src/test/functional/com/mongodb/client/unified/UnifiedTest.java

```diff
@@ -288,11 +294,76 @@ private OperationResult executeOperation(final BsonDocument operation) {
+            case "loop":
+                return loop(operation);
             default:
                 throw new UnsupportedOperationException("Unsupported test operation: " + name);
         }
     }
 
+    private OperationResult loop(final BsonDocument operation) {
+        BsonDocument arguments = operation.getDocument("arguments");
+
+        int numIterations = 0;
+        int numSuccessfulOperations = 0;
+        boolean storeFailures = arguments.containsKey("storeFailuresAsEntity");
+        boolean storeErrors = arguments.containsKey("storeErrorsAsEntity");
+        BsonArray failureDescriptionDocuments = new BsonArray();
+        BsonArray errorDescriptionDocuments = new BsonArray();
+
+        while (!terminateLoop()) {
+            for (BsonValue cur : arguments.getArray("operations")) {
+                try {
+                    assertOperation(cur.asDocument());
+                    numSuccessfulOperations++;
+                } catch (AssertionError e) {
+                    if (storeFailures) {
+                        failureDescriptionDocuments.add(createDocumentFromException(e));
+                    } else if (storeErrors) {
+                        errorDescriptionDocuments.add(createDocumentFromException(e));
+                    } else {
+                        throw e;
+                    }
+                    break;
+                } catch (RuntimeException e) {
+                    if (storeErrors) {
+                        errorDescriptionDocuments.add(createDocumentFromException(e));
+                    } else if (storeFailures) {
+                        failureDescriptionDocuments.add(createDocumentFromException(e));
+                    } else {
+                        throw e;
+                    }
+                    break;
+                }
+            }
+            numIterations++;
+        }
+
+        if (arguments.containsKey("storeSuccessesAsEntity")) {
+            entities.addSuccessCount(arguments.getString("storeSuccessesAsEntity").getValue(), numSuccessfulOperations);
+        }
+        if (arguments.containsKey("storeIterationsAsEntity")) {
+            entities.addIterationCount(arguments.getString("storeIterationsAsEntity").getValue(), numIterations);
+        }
+        if (storeFailures) {
+            entities.addFailureDocuments(arguments.getString("storeFailuresAsEntity").getValue(), failureDescriptionDocuments);
+        }
+        if (storeErrors) {
+            entities.addErrorDocuments(arguments.getString("storeErrorsAsEntity").getValue(), errorDescriptionDocuments);
+        }
+
+        return OperationResult.NONE;
+    }
+
+    private BsonDocument createDocumentFromException(final Throwable throwable) {
+        return new BsonDocument("error", new BsonString(throwable.toString()))
+                .append("time", new BsonDouble(System.currentTimeMillis() / 1000.0));
+    }
+
+    protected boolean terminateLoop() {
```

Comment:
Overriding this method is how the workload executor will force loop termination.